packages/dd-trace/test/debugger/devtools_client/condition.spec.js

```diff
@@ -0,0 +1,261 @@
+    { collection: ['foo', 'bar', ''] },
+    ['foo', 'bar']
+  ],
+  [
+    { filter: [{ ref: 'collection' }, { not: { isEmpty: { ref: '@it' } } }] },
+    { collection: new Set(['foo', 'bar', '']) },
+    ['foo', 'bar']
+  ],
+  [
+    { filter: [{ ref: 'collection' }, { not: { isEmpty: { ref: '@value' } } }] },
+    { collection: { 1: 'foo', 2: 'bar', 3: '' } },
+    { 1: 'foo', 2: 'bar' }
+  ],
+  [
+    { filter: [{ ref: 'collection' }, { not: { isEmpty: { ref: '@key' } } }] },
+    { collection: { foo: 1, bar: 2, '': 3 } },
+    { foo: 1, bar: 2 }
+  ],
+  [{ contains: [{ ref: 'str' }, 'world'] }, { str: 'hello world' }, true],
+  [{ contains: [{ ref: 'str' }, 'missing'] }, { str: 'hello world' }, false],
+  [{ contains: [{ ref: 'arr' }, 'foo'] }, { arr: ['foo', 'bar'] }, true],
+  [{ contains: [{ ref: 'arr' }, 'missing'] }, { arr: ['foo', 'bar'] }, false],
+  [{ contains: [{ ref: 'arr' }, 'foo'] }, { arr: overloadMethod(['foo', 'bar'], 'includes') }, true],
+  // TODO: Test TypedArray
+  [{ contains: [{ ref: 'set' }, 'foo'] }, { set: new Set(['foo', 'bar']) }, true],
+  [{ contains: [{ ref: 'set' }, 'missing'] }, { set: new Set(['foo', 'bar']) }, false],
+  [{ contains: [{ ref: 'set' }, 'foo'] }, { set: overloadMethod(new Set(['foo', 'bar']), 'has') }, true],
+  [{ contains: [{ ref: 'wset' }, { ref: 'key' }] }, { key: weakKey, wset: new WeakSet([weakKey]) }, true],
+  [{ contains: [{ ref: 'wset' }, { ref: 'key' }] }, { key: {}, wset: new WeakSet([weakKey]) }, false],
+  [
+    { contains: [{ ref: 'wset' }, { ref: 'key' }] },
+    { key: weakKey, wset: overloadMethod(new WeakSet([weakKey]), 'has') },
+    true
+  ],
+  [{ contains: [{ ref: 'map' }, 'foo'] }, { map: new Map([['foo', 'bar']]) }, true],
+  [{ contains: [{ ref: 'map' }, 'missing'] }, { map: new Map([['foo', 'bar']]) }, false],
+  [{ contains: [{ ref: 'map' }, 'foo'] }, { map: overloadMethod(new Map([['foo', 'bar']]), 'has') }, true],
+  [{ contains: [{ ref: 'wmap' }, { ref: 'key' }] }, { key: weakKey, wmap: new WeakMap([[weakKey, 'bar']]) }, true],
+  [{ contains: [{ ref: 'wmap' }, { ref: 'key' }] }, { key: {}, wmap: new WeakMap([[weakKey, 'bar']]) }, false],
+  [
+    { contains: [{ ref: 'wmap' }, { ref: 'key' }] },
+    { key: weakKey, wmap: overloadMethod(new WeakMap([[weakKey, 'bar']]), 'has') },
+    true],
+  [
+    { contains: [{ ref: 'obj' }, 'foo'] },
+    { obj: { foo: 'bar' } },
+    new TypeError('Variable obj does not support contains')
+  ],
+  [
+    { contains: [{ ref: 'obj' }, 'missing'] },
+    { obj: { foo: 'bar' } },
+    new TypeError('Variable obj does not support contains')
+  ],
+  // TODO: Ensure there's no side-effects due to Symbol.match
+  [{ matches: [{ ref: 'foo' }, '[0-9]+'] }, { foo: '42' }, true],
+  [{ matches: [{ ref: 'foo' }, { ref: 'regex' }] }, { foo: '42', regex: /[0-9]+/ }, true],
+  [{ matches: [{ ref: 'foo' }, { ref: 'regex' }] }, { foo: '42', regex: overloadMethod(/[0-9]+/, 'test') }, true],
```

Comment:
lets add test when foo is not a string.
and then maybe also overloadMethod of match on foo.